src/backend/Functions/FunctionsDIConfiguration.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DependencyInjection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Fx.Portability.Azure.Storage;
+using Microsoft.Fx.Portability.ObjectModel;
+using Microsoft.WindowsAzure.Storage;
+using System;
+
+namespace Functions
+{
+    public class InjectConfiguration : IExtensionConfigProvider
+    {
+        public virtual void Initialize(ExtensionConfigContext context)
+        {
+            var services = new ServiceCollection();
+            RegisterServices(services);
+            var serviceProvider = services.BuildServiceProvider(true);
+
+            context
+                .AddBindingRule<InjectAttribute>()
+                .Bind(new InjectBindingProvider(serviceProvider));
+
+            var registry = context.Config.GetService<IExtensionRegistry>();
+            var filter = new ScopeCleanupFilter();
+            registry.RegisterExtension(typeof(IFunctionInvocationFilter), filter);
+            registry.RegisterExtension(typeof(IFunctionExceptionFilter), filter);
+        }
+
+        private static void RegisterServices(IServiceCollection services)
+        {
```

Comment:
I wanted to do per-project service registration. However, for unknown reason,
Azure Functions doesn't load this class, even though the source code is mostly
the same as the other one in the DI project...

Reply:
I saw this: https://stackoverflow.com/a/41127737 with functions being unable to load a shared library. Would this fix help consolidate that code?

Reply:
It doesn't look like related. The reference to the class library project works fine. The InjectConfiguration class defined in the reference is getting loaded, but the class (FunctionsDIConfiguration) defined in the Azure Function project is not (I added it to the PR for discussion of the problem)

---
In reply to: [178133415](https://github.com/Microsoft/dotnet-apiport/pull/598#discussion_r178133415) [](ancestors = 178133415)

Reply:
Removed this file. Open [https://devdiv.visualstudio.com/DevDiv/NET%20CAT/_workitems/edit/594906](https://devdiv.visualstudio.com/DevDiv/NET%2520CAT/_workitems/edit/594906) for possible DI improvement.

---
In reply to: [177563482](https://github.com/Microsoft/dotnet-apiport/pull/598#discussion_r177563482) [](ancestors = 177563482)